src/doc/unstable-book/src/compiler-flags/embed-source.md

```diff
@@ -0,0 +1,10 @@
+# `embed-source`
+
+This flag controls whether the compiler embeds the program source code text into
+object debug info section. It takes one of the following values:
+
+* `y`, `yes`, `on` or `true`: put source code in debug info.
+* `n`, `no`, `off`, `false` or no value: omit source code from debug info (the default).
+
+`-C embed-source` requires DWARFv5 and is only supported with the LLVM backend.
```

Comment:
```suggestion
`-Z embed-source` requires DWARFv5 and is only supported with the LLVM backend.
```

Reply:
Also it can probably be implemented in cg_clif too. You don't have to do that in this PR, and it may require a gimli PR first depending on how exactly the source embedding is encoded.